geometric_kernels/sampling/samplers.py

```diff
@@ -27,23 +26,23 @@ def sample_at(
 
     num_features = B.shape(features)[-1]
 
-    key, random_weights = B.randn(key, float_like(X), num_features, s)  # [M, S]
+    key, random_weights = B.randn(key, B.dtype(params["lengthscale"]), num_features, s)  # [M, S]
 
     random_sample = B.matmul(features, random_weights)  # [N, S]
 
     return key, random_sample
 
 
 def sampler(
-    feature_map: FeatureMap, s: Optional[int] = 1, **kwargs
+    feature_map: FeatureMap, s: Optional[int] = 1, key=None, **kwargs
```

Comment:
What is the reason to have the `key` fixed here, as opposed to have it as a parameter for the sample itself?

Reply:
Removed it. To be honest, I don't remember why I added it in the first place, hopefully for no reason.